skorch/callbacks/logging.py

```diff
@@ -176,6 +176,18 @@ def __init__(
         self.keys_ignored = keys_ignored
         self.base_namespace = base_namespace
 
+        self._log_integration_version()
+
+    def _log_integration_version(self) -> None:
+        try:
+            from importlib.metadata import version
+            skorch_version = version("skorch")
+        except ImportError:
+            import pkg_resources
+            skorch_version = pkg_resources.get_distribution('skorch').version
```

Comment:
Is there any advantage over this method vs just `skorch.__version__`? I know that using these functions is more generic, but since we're only interested in skorch here, and the `__version__` attribute is not going to change, we can safely use it here, right?

Reply:
Got circular imports when trying what you suggest. Look at how this is implemented in `skorch/skorch/__init__.py`. Maybe I missed something

Reply:
Ah yes, makes sense, `__init__.py` imports `callbacks`, which imports the `NeptuneLogger`. It would, however, work, if the skorch import is done locally inside the method, right?

One more thing I noticed: Could you please move the `_log_integration_version` call from `__init__` to `initialize`?

Reply:
Done ;)

Reply:
Shall I also add `__version__` to `__all__` in the `skorch/__init__.py` file to avoid IDE complaints?

Reply:
> Shall I also add `__version__` to `__all__` in the `skorch/__init__.py` file to avoid IDE complaints?

Hmm, what does the IDE complain about, that `__version__` does not exist? Anyway, I'd rather have it in a separate PR.

Reply:
`'__version__' is not declared in __all__`

Reply:
Really not sure about the advantage of adding `__version__` to `__all__`. Maybe if I see a convincing argument :). Anyway, this PR looks good, thx!